### PR TITLE
Add options related to the table <div>

### PR DESCRIPTION
**Pull request by rich-iannone**
This PR adds the ability to modify the width and height of the enclosing table `<div>` for HTML output (though the `container.width` and `container.height` arguments in `tab_options()`). To make this work better with heights less than the actual table height, the `overflow-y` option has been set as `auto` to enable vertical scrolling (there is already the `overflow-x` option set for horizontal scrolling).

This also adds the `table.align` argument (again, in `tab_options()`). It's now possible to `left`, `center` (default), and `right` align tables within the `<div>`.

Here is an example:

```r
library(gt)
library(tidyverse)

pizzaplace %>%
  dplyr::group_by(type, size) %>%
  dplyr::summarize(
    sold = n(),
    income = sum(price)
  ) %>%
  gt(rowname_col = "size") %>%
  tab_header(title = "Pizzas Sold in 2015") %>%
  fmt_number(
    columns = vars(sold),
    decimals = 0,
    use_seps = TRUE
  ) %>%
  fmt_currency(
    columns = vars(income),
    currency = "USD"
  ) %>%
  summary_rows(
    groups = TRUE,
    columns = vars(sold),
    fns = list(`Group Total` = "sum"),
    formatter = fmt_number,
    decimals = 0,
    use_seps = TRUE
  ) %>%
  summary_rows(
    groups = TRUE,
    columns = "income",
    fns = list(`Group Total` = "sum"),
    formatter = fmt_currency,
    currency = "USD"
  ) %>%
  grand_summary_rows(
    columns = "income",
    fns = list(`All Sales` = "sum"),
    formatter = fmt_currency,
    currency = "USD"
  ) %>%
  tab_options(
    summary_row.background.color = "#ACEACE",
    grand_summary_row.background.color = "steelblue",
    row_group.background.color = "#FFEFDB",
    table.width = pct(90),
    container.height = px(300),
    container.width = px(600),
    table.align = "right"
  )
```

Here is the table in a browser with the Web Inspector showing the layout:

gt_div_options

Fixes https://github.com/rstudio/gt/issues/206.
Fixes https://github.com/rstudio/gt/issues/168.

### CODE REVIEW

File: R/print.R [hunk 1]
```diff
@@ -52,7 +59,14 @@ as.tags.gt_tbl <- function(x, ...) {
   # Attach the dependency to the HTML table
   html_tbl <- htmltools::tagList(
     htmltools::tags$style(htmltools::HTML(css)),
-    tags$div(id = id, style = htmltools::css(`overflow-x` = "auto"), htmltools::HTML(html_table))
+    tags$div(
+      id = id,
+      style = htmltools::css(
+        `overflow-x` = "auto",
+        `overflow-y` = "auto",
```
Comment (schloerke) [inline R/print.R]: because these are being set locally and how priority works within css, they should be `tab_options` too

File: R/shiny.R [hunk 2]
```diff
@@ -1,26 +1,44 @@
...
 #'   call to [gt_output()] when `render_gt` is used in an interactive R Markdown
 #'   document.
-#' @seealso \link{gt_output}()
 #' @family Shiny functions
 #' @importFrom htmltools resolveDependencies findDependencies doRenderTags
 #' @export
 render_gt <- function(expr,
+                      width = NULL,
+                      height = NULL,
+                      align = NULL,
```
Comment (schloerke) [inline R/shiny.R]: I'd add an example or two showing basic usage and custom usage

File: R/shiny.R [hunk 3]
```diff
@@ -1,26 +1,44 @@
...
                       env = parent.frame(),
                       quoted = FALSE,
                       outputArgs = list()) {
 
+  # TODO: width should mean container width
```
Comment (schloerke) [inline R/shiny.R]: To resolve

File: R/shiny.R [hunk 4]
```diff
@@ -1,26 +1,44 @@
...
+  # TODO: include example where the table width overflows the container width
```
Comment (schloerke) [inline R/shiny.R]: To resolve

File: R/tab_options.R [hunk 5]
```diff
@@ -203,21 +214,24 @@ tab_options <- function(data,
...
 
-  # table.font.size
-  if (!is.null(table.font.size)) {
+  # container.width
+  if (!is.null(container.width)) {
 
-    if (is.numeric(table.font.size)) {
-      table.font.size <- paste0(table.font.size, "px")
+    if (is.numeric(container.width)) {
+      container.width <- paste0(container.width, "px")
```
Comment (schloerke) [inline R/tab_options.R]: use `px`?

File: R/tab_options.R [hunk 6]
```diff
@@ -203,21 +214,24 @@ tab_options <- function(data,
...
 
-  # table.background.color
-  if (!is.null(table.background.color)) {
+  # container.height
+  if (!is.null(container.height)) {
 
-    opts_df <- opts_df_set(
-      opts_df, "table_background_color", table.background.color)
+    if (is.numeric(container.height)) {
+      container.height <- paste0(container.height, "px")
```
Comment (schloerke) [inline R/tab_options.R]: use `px`?

File: R/tab_options.R [hunk 7]
```diff
@@ -230,6 +244,42 @@ tab_options <- function(data,
...
+      opts_df <- opts_df_set(opts_df, "margin_left", "auto")
+      opts_df <- opts_df_set(opts_df, "margin_right", "0")
+    }
+  }
+
+  # table.font.size
+  if (!is.null(table.font.size)) {
+
+    if (is.numeric(table.font.size)) {
+      table.font.size <- paste0(table.font.size, "px")
```
Comment (schloerke) [inline R/tab_options.R]: use `px`?

File: R/tab_options.R [hunk 8]
```diff
@@ -230,6 +244,42 @@ tab_options <- function(data,
     opts_df <- opts_df_set(opts_df, "table_width", table.width)
   }
 
+  # table.align
+  if (!is.null(table.align)) {
+
+    if (table.align == "center") {
+      opts_df <- opts_df_set(opts_df, "margin_left", "auto")
```
Comment (schloerke) [inline R/tab_options.R]: being able to set a `margin_left` / right but not being able to set it directly seems off to me.

Comment (schloerke) [reply R/tab_options.R]: I can't think of a _nice_ solution without throwing warning messages if certain values have already been set.  Almost as if `table.align` is a _super_ value that will set multiple table options.  If some of those table options have already been set, maybe display a warning?  

But that solution seems to not be aligned with ggplot2 theme ideas.  Hmmm

File: R/print.R (hunk 1 above)
Comment (rich-iannone) [reply R/print.R]: Good idea. Should we call them `table.overflow.x` and `table.overflow.y`?

File: R/shiny.R (hunk 3 above)
Comment (rich-iannone) [reply R/shiny.R]: This is now resolved.

File: R/print.R (hunk 1 above)
Comment (schloerke) [reply R/print.R]: `container.overflow.x`?

File: R/tab_options.R (hunk 5 above)
Comment (rich-iannone) [reply R/tab_options.R]: Now resolved by using `px()` throughout.

File: R/tab_options.R (hunk 8 above)
Comment (rich-iannone) [reply R/tab_options.R]: The way it's set up now is to have `table.align` stomp over `margin_left` and `margin_right`. I can't help but wonder that there should be a `table_align` setting that stored and takes precedence over any setting of `margin_left` or `margin_right`.

File: R/shiny.R (hunk 4 above)
Comment (rich-iannone) [reply R/shiny.R]: Both now resolved.

File: R/print.R (hunk 1 above)
Comment (rich-iannone) [reply R/print.R]: Used `container.overflow.*`.

**Pull request merged**